pandas/core/indexes/interval.py

```diff
@@ -484,13 +493,34 @@ def from_arrays(cls, left, right, closed='right', name=None, copy=False,
                       closed='right',
                       dtype='interval[int64]')
 
+        If you want to segment different groups of people based on
+        ages, you can apply the method as follows:
+
+        >>> ages = pd.IntervalIndex.from_arrays([0, 2, 13],
+        ...                                     [2, 13, 19], closed='left')
+        >>> ages
+        IntervalIndex([[0, 2), [2, 13), [13, 19)]
+              closed='left',
+              dtype='interval[int64]')
+        >>> s = pd.Series(['baby', 'kid', 'teen'], ages)
+        >>> s
+        [0, 2)      baby
+        [2, 13)      kid
+        [13, 19)    teen
+        dtype: object
+
+        Notes
+        -----
+        Each element of `left` must be smaller or equal to the `right` element
+        at the same position, ie, ``left[i] <= right[i]``.
```

Comment:
Can you also add that all elements within each array  should have similar type?  For example, the following raises on master since the mixed types result in `object` subtype (works on 0.22.0, but will be disallowed in the next release):
```python
In [2]: pd.__version__
Out[2]: '0.23.0.dev0+482.gc3d491a'

In [3]: left = [1, pd.Timestamp('20180101')]

In [4]: right = [2, pd.Timestamp('20180201')]

In [5]: pd.IntervalIndex.from_arrays(left, right)
---------------------------------------------------------------------------
TypeError: category, object, and string subtypes are not supported for IntervalIndex
```